barretenberg/cpp/src/barretenberg/smt_verification/circuit/circuit.hpp

```diff
@@ -168,6 +186,168 @@ void Circuit<FF>::univariate_handler(bb::fr q_m, bb::fr q_1, bb::fr q_2, bb::fr
     }
 }
 
+/**
+ * @brief Relaxes univariate logic constraints(AND/XOR).
+ *
+ * @param cursor current position
+ * @return next position
+ */
+template <typename FF> size_t Circuit<FF>::logic_constraint_handler(size_t cursor)
+{
+    size_t beg = 1;
+    size_t end = 127;
+    size_t p = 0;
+    auto res = static_cast<size_t>(-1);
+
+    bool xor_flag = true;
+    bool and_flag = true;
+    bool logic_flag = true;
+    CircuitProps xor_props;
+    CircuitProps and_props;
+
+    bool stop_flag = false;
+
+    while (beg <= end) {
+        p = (end + beg) / 2;
+
+        xor_props = get_standard_logic_circuit(p * 2, true);
```

Comment:
Could you please add comments here? The fact that you are constructing circuits and then comparing the value is not intuitive (also you are constructing the circuits every time. It would probably be better to cache them after 1 time)